.github/workflows/build.yml

```diff
@@ -26,192 +26,308 @@ concurrency:
   group: ${{ github.workflow }}-${{ github.ref }}
   cancel-in-progress: true
 
+env:
+  # Adapted from docker.yml
+  TOX_ENV:                "docker-${{ github.event.inputs.platform || 'ubuntu-jammy-standard' }}-incremental"
+  BUILD_IMAGE:            "localhost:5000/${{ github.repository }}/sage-${{ github.event.inputs.platform || 'ubuntu-jammy-standard' }}-with-targets:ci"
+  FROM_DOCKER_REPOSITORY: "ghcr.io/sagemath/sage/"
+  FROM_DOCKER_TARGET:     "with-targets"
+  FROM_DOCKER_TAG:        ${{ github.event.inputs.docker_tag || 'dev'}}
+  EXTRA_CONFIGURE_ARGS:   --enable-fat-binary
+
 jobs:
-  get_ci_fixes:
+  build:
     runs-on: ubuntu-latest
```

Comment:
How about
```
test-new  test-mod  test-long
```
instead of
```
build     mod       test-long
```
which are obscure now?

In each job, a docker image is built. What are shared between these buildings? Layers?

Reply:
Ah, current working directory is shared between the containers. Right?

Reply:
Nothing is shared explicitly between the different jobs. Per PR description:
> The steps "again builds incrementally" actually use the GH Actions cache, https://docs.docker.com/build/ci/github-actions/cache/#cache-backend-api. When nothing is cached and the 3 workflows are launched in parallel, they may each build the same thing. But when there's congestion that leads to the workflows to be run serially, the 2nd and 3rd workflow will be able to use the cache from the 1st workflow. This elasticity may be helpful in reducing congestion.

Reply:
> How about
> 
> ```
> test-new  test-mod  test-long
> ```

Thanks. I've renamed it as suggested

Reply:
The the workflow may take longer than before, because each job builds the image and the container anew?

Reply:
Only when there is so much pressure on the cache that the layers are no longer there when the next of these jobs is started. Haven't observed that yet.

Reply:
So this
```
          cache-from: type=gha
          cache-to:   type=gha,mode=max
```
sets up the cache? If this cache is local to a job, then how could this allow layers to be shared (via the cache) between three docker image buildings...?

Reply:
No, the cache is global.

Reply:
I see. That's the reason.